src/app/clusters/user-label-server/tests/TestUserLabelCluster.cpp

```diff
@@ -0,0 +1,186 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <pw_unit_test/framework.h>
+
+#include <app/clusters/user-label-server/user-label-cluster.h>
+
+#include <app/clusters/testing/AttributeTesting.h>
+#include <app/server-cluster/AttributeListBuilder.h>
+#include <app/server-cluster/DefaultServerCluster.h>
+#include <app/server-cluster/testing/TestServerClusterContext.h>
+#include <clusters/UserLabel/Attributes.h>
+#include <clusters/UserLabel/Metadata.h>
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::UserLabel;
+using namespace chip::app::Clusters::UserLabel::Attributes;
+
+namespace {
+
+struct TestUserLabelCluster : public ::testing::Test
+{
+    static void SetUpTestSuite() { ASSERT_EQ(Platform::MemoryInit(), CHIP_NO_ERROR); }
+    static void TearDownTestSuite() { Platform::MemoryShutdown(); }
+};
+
+constexpr size_t kUserLabelFixedClusterCount = 2;
+
+class UserLabelClusterTest
+{
+    UserLabelCluster userLabel;
+
+public:
+    template <typename... Args>
+    UserLabelClusterTest(Args &&... args) : userLabel(std::forward<Args>(args)...)
+    {}
+
+    template <typename F>
+    void Check(F check)
+    {
+        check(userLabel);
```

Comment:
Please drop this complexity: this adds a template with varargs that uses a template check for tests (and that loses ASSERT_ as exit points, so it is incomplete).

We just want a cluster test, so tests can create a `UserLabelCluster cluster(id)` and run the test.
We also do not need a loop to test everything twice. This should make the tests simpler.